From 64696f065c87fac32b270a6e864052709193e98c Mon Sep 17 00:00:00 2001
From: Phil Edworthy <phil.edworthy@renesas.com>
Date: Tue, 15 Nov 2016 11:25:12 +0000
Subject: [PATCH] dfu: dfu_sf: Calculate 'remaining' partition size

If a partition size is zero, automatically fill it in with the
'remaining' of the flash.

Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
---
 drivers/dfu/dfu_sf.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/dfu/dfu_sf.c b/drivers/dfu/dfu_sf.c
index 51311c7..d6ffaab 100644
--- a/drivers/dfu/dfu_sf.c
+++ b/drivers/dfu/dfu_sf.c
@@ -161,6 +161,12 @@ int dfu_fill_entity_sf(struct dfu_entity *dfu, char *devstr, char *s)
 		dfu->data.sf.start = simple_strtoul(s, &s, 16);
 		s++;
 		dfu->data.sf.size = simple_strtoul(s, &s, 16);
+		if (dfu->data.sf.size == 0) {
+			dfu->data.sf.size = dfu->data.sf.dev->size -
+							dfu->data.sf.start;
+			debug("DFU:%s calculated size is 0x%llx bytes\n",
+				dfu->name, dfu->data.sf.size);
+		}
 	} else {
 		printf("%s: Memory layout (%s) not supported!\n", __func__, st);
 		spi_flash_free(dfu->data.sf.dev);
-- 
2.7.4

